frontend/src/components/BookTable.tsx

```diff
@@ -0,0 +1,624 @@
+    return vals.split(',');
+  }
+  if (/^#([a-f0-9]{3}){1,2}$/.test(c)) {
+    if (c.length == 4) {
+      c = '#' + [c[1], c[1], c[2], c[2], c[3], c[3]].join('');
+    }
+    c = '0x' + c.substring(1);
+    return [(c >> 16) & 255, (c >> 8) & 255, c & 255];
+  }
+  return '';
+}
+
+function mixColors(baseColor: string, accentColor: string, point: number) {
+  const baseRGB = hexToRgb(baseColor);
+  const accentRGB = hexToRgb(accentColor);
+  const redDiff = accentRGB[0] - baseRGB[0];
+  const red = baseRGB[0] + redDiff * point;
+  const greenDiff = accentRGB[1] - baseRGB[1];
+  const green = baseRGB[1] + greenDiff * point;
+  const blueDiff = accentRGB[2] - baseRGB[2];
+  const blue = baseRGB[2] + blueDiff * point;
+  return `rgb(${Math.round(red)}, ${Math.round(green)}, ${Math.round(blue)}, ${0.7 + point * 0.3})`;
+}
+
+function localizeDataGrid() {
+  const { t } = useTranslation();
+  return {
+    MuiTablePagination: { labelRowsPerPage: t('Orders per page:') },
+    noRowsLabel: t('No rows'),
+    noResultsOverlayLabel: t('No results found.'),
+    errorOverlayDefaultLabel: t('An error occurred.'),
+    toolbarColumns: t('Columns'),
+    toolbarColumnsLabel: t('Select columns'),
+    columnsPanelTextFieldLabel: t('Find column'),
+    columnsPanelTextFieldPlaceholder: t('Column title'),
+    columnsPanelDragIconLabel: t('Reorder column'),
+    columnsPanelShowAllButton: t('Show all'),
+    columnsPanelHideAllButton: t('Hide all'),
+    filterPanelAddFilter: t('Add filter'),
+    filterPanelDeleteIconLabel: t('Delete'),
+    filterPanelLinkOperator: t('Logic operator'),
+    filterPanelOperators: t('Operator'),
+    filterPanelOperatorAnd: t('And'),
+    filterPanelOperatorOr: t('Or'),
+    filterPanelColumns: t('Columns'),
+    filterPanelInputLabel: t('Value'),
+    filterPanelInputPlaceholder: t('Filter value'),
+    filterOperatorContains: t('contains'),
+    filterOperatorEquals: t('equals'),
+    filterOperatorStartsWith: t('starts with'),
+    filterOperatorEndsWith: t('ends with'),
+    filterOperatorIs: t('is'),
+    filterOperatorNot: t('is not'),
+    filterOperatorAfter: t('is after'),
+    filterOperatorOnOrAfter: t('is on or after'),
+    filterOperatorBefore: t('is before'),
+    filterOperatorOnOrBefore: t('is on or before'),
+    filterOperatorIsEmpty: t('is empty'),
+    filterOperatorIsNotEmpty: t('is not empty'),
+    filterOperatorIsAnyOf: t('is any of'),
+    filterValueAny: t('any'),
+    filterValueTrue: t('true'),
+    filterValueFalse: t('false'),
+    columnMenuLabel: t('Menu'),
+    columnMenuShowColumns: t('Show columns'),
+    columnMenuFilter: t('Filter'),
+    columnMenuHideColumn: t('Hide'),
+    columnMenuUnsort: t('Unsort'),
+    columnMenuSortAsc: t('Sort by ASC'),
+    columnMenuSortDesc: t('Sort by DESC'),
+    columnHeaderFiltersLabel: t('Show filters'),
+    columnHeaderSortIconLabel: t('Sort'),
+    booleanCellTrueLabel: t('yes'),
+    booleanCellFalseLabel: t('no'),
+  };
```

Comment:
this constant might be just declared on the component, it's not necessary to have a function